earthfile2llb/converter.go

```diff
@@ -78,6 +79,8 @@ type Converter struct {
 	directDeps          []*states.SingleTarget
 	buildContextFactory llbfactory.Factory
 	cacheContext        pllb.State
+	runOpts             []llb.RunOption
+	persistentCacheDirs map[string]bool // Treated as a "set" of cache directories to persist in the image
```

Comment:
@alexcb let me know what you think of this approach. I've found it to be an idiomatic way in Go to achieve set-like behavior and/or get O(1) lookup without looping through an array

Reply:
That sounds good, we do this in other places of the code too. I've also seen it as `map[string]struct{}`, which makes it even more clear that it's a set as there's no associated value with it. And according to https://itnext.io/set-in-go-map-bool-and-map-struct-performance-comparison-5315b4b107b using `struct{}` is actually faster and reduces memory. (not like memory is a concern when I suspect N<10 in most cases).